tox.ini

```diff
@@ -47,13 +48,16 @@ passenv =
     SSH_CONFIG_FILE
     TEST_HOSTS_LIST
     ISO_MOUNTPOINT
+setenv =
+    VAGRANT_CWD={toxinidir}
+commands_pre =
```

Comment:
Are commands specified in `commands_pre` run for each invocation of this environment? If not (e.g. only on env creation), this might become problematic.

Reply:
So it seems they are. However, we run it even if passing an `SSH_CONFIG_FILE` env variable - which we do in the CI, where there is no Vagrant. As discussed, prefixing the command with a dash `-` should ignore the return code.

_Or_, you could test the environment variable in `commands_pre`, e.g.:
```
whitelist_externals =
    {[testenv]whitelist_externals}
    vagrant
...
commands_pre =
    [[ -z "$SSH_CONFIG_FILE" ]] && vagrant ssh-config bootstrap > {envdir}/vagrant_ssh_config
```

Reply:
yes it is run each invocation of the env